src/paste.rs

```diff
@@ -230,9 +232,16 @@ impl Paste {
         let bytes_checksum = util::sha256_digest(&*bytes)?;
         self.data = bytes;
         if !config.paste.duplicate_files.unwrap_or(true) && expiry_date.is_none() {
-            if let Some(file) =
-                Directory::try_from(config.server.upload_path.as_path())?.get_file(bytes_checksum)
-            {
+            let upload_path = config.server.upload_path.clone();
+
+            let directory =
+                match spawn_blocking(move || Directory::try_from(upload_path.as_path())).await {
+                    Ok(Ok(d)) => d,
+                    Ok(Err(e)) => return Err(error::ErrorInternalServerError(e)),
+                    Err(e) => return Err(error::ErrorInternalServerError(e)),
+                };
```

Comment:
Why are we spawning a task here?

Reply:
`Directory::try_from` uses `glob`, which is strictly synchronous. I could move the spawn into `try_from`, but there's no `AsyncTryFrom` trait, so spwaning at the call site was the best I could do.